### ISSUE TITLE
institute_id vs institution_id in filename

### ISSUE THREAD

**Issue by klauswyser** [PCMDI/cmor#175]
According to the CMIP6_CV.json file the attribute for the institute id is called <institution_id>. However, when I process my model output I get a wrong directory structure, e.g.

 $outpath/CMIP6/institute_idEC-Earth-3-HR/CMIP/historical/ 

where <institution_id> has not been used and instead the term "institute_id" appears. Not sure if this is a specific problem of CMIP6 or a more general problem of the CMOR3 library.

**Comment by dnadeau4** [comment]
Did you set your `insitution_id` in your `CMOR input JSON dataset file`?  
It should look like this:
https://github.com/PCMDI/cmor/blob/master/Test/common_user_input.json#L45

It needs to match one of the institution registered:
https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_institution_id.json#L2-L33

**Comment by klauswyser** [comment]
Yes, I did set <institution_id> correctly, and it's in the list of registered institutions:

``    "institution_id":               "EC-Earth-Consortium",``

**Comment by dnadeau4** [comment]
What version of CMOR are you using?

Can you `diff` with the  file in attachment?  I was able to create the right directory.
`CMIP6/CMIP6/CMIP/EC-Earth-Consortium/EC-Earth3-HR/historical/r11i1p1f1/Omon/masso/gr/v20170517`
[common_user_input.json.txt](https://github.com/PCMDI/cmor/files/1009247/common_user_input.json.txt)

**Comment by dnadeau4** [comment]
Can you verify your path template as well?  CMOR will loop on all tags and replace the tags for its attribute value you have defined. 

```
line  71            "output_path_template":    "<mip_era><activity_id><institution_id><source_id><experiment_id><_member_id><table><variable_id><grid_label><version>",
```

**Comment by klauswyser** [comment]
Version is a good question, I am little bit confused myself. When I updated the cmor3 library yesterday I got updates from master and 3.4.4. The INSTALL file says 3.2.3 and so does cmor.h. However, the RELEASE-NOTES say 3.2.2, and the resulting netCDF file also says 3.2.2 (produced with python). Which version am I using? Anyway, I did an update yesterday before testing, so it should be reasonably fresh.

I compared you common_input.json file against my metadata-input.json, it seems that my metadata-input.json is coming from an older version and we need to bring it up to date.

My metadata file has neither <output_path_template> nor <output_file_template>. Are they really needed? I thought that the default path in CMOR3 is set correctly for CMIP6, and the <outpath_*_templates> are only needed if a user wants to deviate from the default. Isn't that the case?

**Comment by ehogan** [comment]
I believe the issue is related to the fact that [CMOR_DEFAULT_PATH_TEMPLATE](https://github.com/PCMDI/cmor/blob/master/include/cmor.h#L255) contains `<institute_id>` rather than `<institution_id>` (which Denis has defined in the JSON dataset file and so overrides the default).

**Comment by klauswyser** [comment]
Indeed, I can find <institute_id> in cmor.h, but is that used for python? Sorry, I am a python newbie with not much experience. Where would I find the corresponding default for python?`I serached through all .py and .json files, nada. 

While we are at it: both fileaname and pathname contain ``member_id``, but this attribute isn't defined in my user_input.json file and therefore my path- and filename contain the string _member_id. Somehow this global internal attribute should be constructed from realisation, forcing, etc, But apparently this isn't happening. Any advice?

**Comment by ehogan** [comment]
CMOR contains only a Python interface to the C code. `cmor.h` is used by the C code, so will, in turn, get used when you access CMOR via the Python interface. I hope that helps?

Good question about `member_id`; I'll let @dnadeau4 answer that one! :)

**Comment by klauswyser** [comment]
Thanks for the explanation, it helps! The question is then if ``institute_id`` will be replaced by ``institution_id`` in future updates of the CMOR library, or do all users have to set the path and file templates explicitely in the common user input json file?

**Comment by ehogan** [comment]
> Thanks for the explanation, it helps! The question is then if `institute_id` will be replaced by `institution_id` in future updates of the CMOR library, or do all users have to set the path and file templates explicitely in the common user input json file?

I think for CMIP6, the idea was for CMOR to use the default value, since that would ensure the file name was correct. @dnadeau4 will (hopefully!) confirm this :)

**Comment by dnadeau4** [comment]
@klauswyser version on conda forge is 3.2.3 and I am working on 3.2.4  Everything should be set to 3.2.3 in the master branch.  I did forget to update the RELEASE-NOTES.  

If your final netCDF4 says 3.2.2 than  you are using 3.2.2.  I change the cmor.h file everytime. 

@ehogan thanks for pointing the DEFAULT template problem.  I just fixed it.  I looked at the spreadsheet ant `institute_id` was changed to `institution_id` from CMIP5 to CMIP6.
https://docs.google.com/document/d/1h0r8RZr_f3-8egBMMh7aqLwy3snpD6_MrDz1q8n5XUk/edit

**Comment by dnadeau4** [comment]
_member_id was introduced by @taylor13 and is setup internally by CMOR.  BTW all internal tags starting with`_` will not be added to the netCDF files.  

Here is the logic
```
if sub_experiment_id = “none”
    _member_id = <variant_label>
else
    _member_id = <sub_experiment_id>-<variant_label>
endif
```

@klauswyser  The [CMOR_DEFAULT_PATH_TEMPLATE](https://github.com/PCMDI/cmor/blob/master/include/cmor.h#L255) should not be overwritten for `CMIP6` users.   `obs4MIPs` asked for a lot of flexibility by overwriting almost everything....

**Comment by taylor13** [comment]
@klauswyser   Here's the document where member_id is defined:   goo.gl/v1drZl

**Comment by klauswyser** [comment]
Thanks, I have read this document before and know how ´´member_id´´ is constructed. If I understand the CMOR library correctly, then ``member_id`` is an internal attribute that should be constructed automatically from the user provided input ``realization``, ``forcing``, etc. It is done like that in PrePARE, but I cannot find something similar within the CMOR library for processing our climate model output, yet the ``member_id`` attribute is used in the output path- and filename. Something missing?

**Comment by klauswyser** [comment]
@dnadeau4 wrote:
> If your final netCDF4 says 3.2.2 than you are using 3.2.2. I change the cmor.h file everytime.

Well, my cmor.h file says
```
#define CMOR_VERSION_MAJOR 3
#define CMOR_VERSION_MINOR 2
#define CMOR_VERSION_PATCH 3
```
yet my netCDF file says
```
                :cmor_version = "3.2.2" ;
```

**Comment by klauswyser** [comment]
@dnadeau4 Sorry for all the trouble, I just realized that the CMOR library in my virtual python environment was not properly updated after I updated CMOR from github, ``make python`` didn't update the CMOR installation and ``pip uninstall CMOR`` didn't help either. After removing manually cmor from the site-packages directory and re-installing the CMOR library, everything works just fine: ``model_id`` is properly defined, and the version is set properly to 3.2.3 in the netCDF file.

Once more sorry for the trouble, and thanks a lot for all the help.
